### ISSUE TITLE
Proposal for streaming with Chunked Transfer Encoding

### ISSUE THREAD

**Issue by subyraman** [channelcat/sanic#466]
I've implemented a proof of concept for chunked transfer encoding in this branch: https://github.com/subyraman/sanic/tree/explore-streams

The user-facing API would be a new response method, `stream`, which would accept a coroutine that writes to the response:

```python
async def streaming_fn(response):
    await response.write('foo')
    await asyncio.sleep(.5)
    await response.write('bar')
    await asyncio.sleep(.5)


@app.route("/")
async def test(request):
    return stream(streaming_fn, content_type='text/plain')
```

The `stream` method would instantiate a new class, `StreamingHTTPResponse`, which would be similar to the existing `HTTPResponse` except it exposes a method `write` which would write content to the transport:

```python
class StreamingHTTPResponse:
    def __init__(self, streaming_fn, status=200, headers=None,
                 content_type='text/plain'):
        self.content_type = content_type
        self.streaming_fn = streaming_fn
        self.status = status
        self.headers = headers or {}
        self._cookies = None

    async def write(self, data):
        ret = "{}\r\n{}\r\n".format(len(data), data).encode()
        self.transport.write(ret)

    async def start_stream(self):
        headers = self.get_headers()
        self.transport.write(headers)
        await self.streaming_fn(self)
```

The user could initialize the `StreamingHTTPResponse` themselves and do things like attach signal handlers which could close the transport on a particular event, but I haven't thought through an event API.

I wanted to get any thoughts; the API pretty closely follows the Django/Flask model for streaming except we can do neat things here asynchronously (like event handling).

**Comment by r0fls** [comment]
Open PR for this here: https://github.com/channelcat/sanic/pull/473

**Comment by agoose77** [comment]
Can this be closed now PR accepted?

**Comment by seemethere** [comment]
Closed for #473
